lib/note-list/index.tsx

```diff
@@ -29,16 +29,37 @@ import {
 } from './decorators';
 import TagSuggestions, { getMatchingTags } from '../tag-suggestions';
 
-import { closeNote } from '../state/ui/actions';
+import actions from '../state/actions';
```

Comment:
> I like knowing the actions that are called in a file from the imports.

Just out of curiosity, what are you looking for in knowing which actions it calls?

> If there are so many actions being called from a component that the import becomes onerous the component is probably doing too much.

Like most of these refactors I'm not trying to change behavior but rather adhere to the existing code as much as possible.

> I guess it makes searching easier for the action as it will only show where it is called. What other arguments are there for doing the import like this?

either way we won't have trouble searching for the action names since in this case they are guaranteed to match their declaration's name (can't do `import { closeNote as closeIt }`) but it's a style thing.

I tend to prefer this style for a few reasons:
 - we get the autocomplete and that's important to me because I don't always know exactly what I need and I like to be able to scan through the available actions
 - consistent code around the app; not a big reason to me but some people like this
 - we don't run into shadowing issues using the `actions.ui.closeNote` calls since we tend to like to name things passed into the component the same thing as the actions. I haven't seen a shadow warning lately so I don't know if this is relevant here based on how we've been coding.

we should be able to make one fell swoop and update all occurrences via a codemod should we decide to settle on one. in the meantime I'm trying to follow suit unless I'm adding a few new imports.

> If you separate them you get import checking in most editors. I guess TypeScript will probably throw an error if the function doesn't exist.

should type check fine either way

Reply:
I've thought a lot about this and I think `import actions from '../state/actions';` is the way to go.  I had a really bad rebase that made me flip to @dmsnell 's way of thinking.

D has also made very valid points here which have also swayed me.